design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,455 @@
+        env:
+        - name: MY_POD_NAME
+          valueFrom:
+            fieldRef:
+              apiVersion: v1
+                fieldPath: metadata.name
+
+        - name: MY_POD_NAMESPACE
+          valueFrom:
+          fieldRef:
+            apiVersion: v1
+            fieldPath: metadata.namespace
+
+        - name: CONJUR_APPLIANCE_URL
+          value:  {{ .Values.secrets-provider-environment.conjurApplianceUrl }}
+
+        - name: CONJUR_AUTHN_URL
+          value:  {{ .Values.secrets-provider-environment.conjurAuthnUrl }}
+
+        - name: CONJUR_ACCOUNT
+          value:  {{ .Values.secrets-provider-environment.serviceAccount }}
+
+        - name: CONJUR_SSL_CERTIFICATE
+       	  valueFrom:
+            configMapKeyRef:
+        	  name: {{ .Values.secrets-provider-environment.conjurSslCertificate }}
+       		  key: ssl-certificate
+
+        - name: DEBUG
+       	  value:  {{ .Values.secrets-provider-environment.debug }}
+
+        - name: CONJUR_AUTHN_LOGIN
+       	  value:  {{ Values.secrets-provider-environment.conjurAuthnLogin }}
+
+        {{- if eq .Values.secrets-provider-environment.rotationSupport "true" }}
+        - name: SYNC_INTERVAL_TIME
+          value: {{ .Values.secrets-provider-environment.syncInternalTime }}
+        {{- end }}
+
+        {{- if .Values.secrets-provider-environment.k8SecretsLabels }}
+        - name: K8S_SECRETS_LABELS
+          value: {{ .Values.secrets-provider-environment.k8sSecretsLabels }}
+        {{- end }}
+```
+
+Note that when we transition to Milestone 2 we will need to keep the following in mind:
+
+`rotationSupport=false` default will need to be updated to`rotationSupport=true`. 
+
+We will need to add `syncIntervalTime=300` (5 minutes) to the default `values.yaml` where `syncIntervalTime` will represent how often the Secrets Provider will attempt to fetch new Conjur secrets. If the customer would like to change this parameter, they will need to do so manually by *reinstalling* *the Secrets Provider Chart*. 
+
+`ttlSecondsAfterFinished` will need to be removed from defaults
+
+All these changes *demands* that the customer manually deletes and installs the latest Helm Charts
+
+#### Packaging Helm Charts
+
+Every Helm project requires a `Chart.yml` file that describes the project. The Secrets Provider `Chart.yaml` wil resemble the following:
+
+```yaml
+apiVersion: v1
+description: A Helm chart for CyberArk Secrets Provider for Kubernetes
+home: https://www.cyberark.com
+icon: https://xebialabs-clients-iglusjax.stackpathdns.com/assets/files/logos/CyberArkConjurLogoWhiteBlue.png
+keywords:
+- security
+- secrets management
+maintainers:
+- email: conj_maintainers@cyberark.com
+  name: Conjur Maintainers
+name: secrets-provider
+version: 1.0.0
+```
+
+Helm Charts are packaged in `tgz` format. 
+
+For each release, we will package our Chart (`helm package secrets-provider`) and the output will be `secrets-provider-<version>.tgz`. This `.tgz` is what we will push to Github.
```

Comment:
will this be automated?

Reply:
We definitely could but IMO there isn't much to automate here just that command and `git push`. Do you disagree?

Reply:
Basically this is what is done in these case now, but it would be great (only if you have a time) to add 'helm package' command to repo's jenkinsfile

Reply:
i think that any process that needs to happen should be automated if possible. we already have some automated process for releases so we can add it there.

Reply:
If I am understanding correctly, during the `publish` step in Jenkins where we are pushing to our Docker registry and RH registry, add the Helm package and git add/commit/push step?

Reply:
helm code is pushed to github registry when you are doing git push. It issues run of jenkins job. One of steps of jenkins job is build, which will execute 'helm package secret-provider-helm' and create secret-provider-helm.tgz. 

You may also try to look how the conjur-oss-helm.tgz is created and to it the same

Reply:
but from my understanding `secret-provider-helm.tgz` is the artifact that needs to be pushed to GH as well

Reply:
>helm code is pushed to github registry when you are doing git push

What is github registry? you mean [github packages](https://github.com/features/packages)?

Reply:
I meant github repo (which will be dedicated for this purpose)

Reply:
as in this secrets-provider-for-k8s repo correct?

Reply:
yes

Reply:
@oburstein-hub @sigalsax I think we should use [Helm Package in GitHub](https://helm.sh/docs/topics/chart_repository/#github-pages-example)